core/commands/dns.go

```diff
@@ -0,0 +1,82 @@
+package commands
+
+import (
+	"io"
+	"strings"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+	namesys "github.com/ipfs/go-ipfs/namesys"
+	util "github.com/ipfs/go-ipfs/util"
+)
+
+var DNSCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "DNS link resolver",
+		ShortDescription: `
+Multihashes are hard to remember, but domain names are usually easy to
+remember.  To create memorable aliases for multihashes, DNS TXT
+records can point to other DNS links, IPFS objects, IPNS keys, etc.
+This command resolves those links to the referenced object.
+`,
+		LongDescription: `
+Multihashes are hard to remember, but domain names are usually easy to
+remember.  To create memorable aliases for multihashes, DNS TXT
+records can point to other DNS links, IPFS objects, IPNS keys, etc.
+This command resolves those links to the referenced object.
+
+For example, with this DNS TXT record:
+
+  ipfs.io. TXT "dnslink=/ipfs/QmRzTuh2Lpuz7Gr39stNr6mTFdqAghsZec1JoUnfySUzcy ..."
+
+The resolver will give:
+
+  > ipfs dns ipfs.io
+  /ipfs/QmRzTuh2Lpuz7Gr39stNr6mTFdqAghsZec1JoUnfySUzcy
+
+And with this DNS TXT record:
+
+  ipfs.ipfs.io. TXT "dnslink=/dns/ipfs.io ..."
+
+The resolver will give:
+
+  > ipfs dns ipfs.io
```

Comment:
On Thu, May 14, 2015 at 09:21:34PM -0700, Juan Batiz-Benet wrote:
> > +  > ipfs dns ipfs.io
> 
> may want to use `foo.ipfs.io`

So two questions:

1. Does it really matter if this example matches up with live DNS
   records?
2. What live DNS links do we have?  I don't see TXT records for
   ipfs.io, ipfs.ipfs.io, or foo.ipfs.io.